src/main/kotlin/no/skatteetaten/aurora/fergus/service/StorageGridService.kt

```diff
@@ -1,20 +1,20 @@
 package no.skatteetaten.aurora.fergus.service
 
-import no.skatteetaten.aurora.fergus.controllers.AuthorizationPayload
-import org.springframework.web.reactive.function.client.WebClient
+import kotlinx.coroutines.reactive.awaitSingle
 import no.skatteetaten.aurora.fergus.FergusException
-import no.skatteetaten.aurora.fergus.ServiceTypes
-import no.skatteetaten.aurora.fergus.TargetService
-
-class StorageGridServiceReactive(
-    @TargetService(ServiceTypes.STORAGEGRID) private val webClient: WebClient
-) : StorageGridService {
-    override suspend fun authorize(authorizationPayload: AuthorizationPayload): AuthorizeResponse? = webClient
-        .post()
-        .uri("/api/v3/authorize")
-        .authorize()
-
-    private suspend inline fun WebClient.RequestHeadersSpec<*>.authorize(): AuthorizeResponse? = null
+import no.skatteetaten.aurora.fergus.controllers.AuthorizationPayload
+import org.openapitools.client.api.AuthApi
+import org.openapitools.client.model.AuthorizeResponse
+import org.openapitools.client.model.Credentials
+import org.springframework.stereotype.Service
+
+@Service
+class StorageGridServiceReactive(private val storageGridAuthApi: AuthApi) : StorageGridService {
+    override suspend fun authorize(
+        authorizationPayload: AuthorizationPayload
+    ): AuthorizeResponse = storageGridAuthApi
```

Comment:
Ikke viktig for denne saken, men her ville jeg nok returnert et objekt som er mer skreddersydd til hva vi faktisk har behov for i Fergus. Det kan løses i AOS-5605 istedet.